### DISCUSSION TITLE
Show/Hide a control in the Anchored Menu

### DISCUSSION THREAD

**Discussion by MichaelWDenney** [kirill-grouchnikov/radiance#495] [Q&A] [answered]
Is there a way to set the visibility of a control in the anchored menu.  Currently I am clearing the icon and text when I don't want the control visible. That almost works in that you can't see it, but you can still click on it if you know where to look.

**Comment by kirill-grouchnikov** [comment]
`JRibbon.removeAnchoredCommand`

**Reply by MichaelWDenney** [reply]
Removing the control isn't really what I need.  Hiding it is a temporary state.  More like a enable/disable really, but, because the functionality is very specific to a limited case, we want to hide the control completely when it is not applicable.  If I remove the control, I can't put it back in the position it came from.

**Reply by MichaelWDenney** [reply]
Btw, I purposely positioned the control at the front of the anchored controls to avoid the controls shifting around when this control is hidden.

**Reply by kirill-grouchnikov** [reply, marked as answer]
Swing doesn't really do "hide". There is no `Container.hide` for example. To stay consistent with Swing container APIs, and to address this specific case, I can add `JRibbon.addAnchoredCommand(CommandButtonProjection projection, int index)`.

**Reply by MichaelWDenney** [reply]
Being able to add the command back in as the first command may work.  I'd have to play with it a bit to see how it looks.  Worth a try though, if you could add the method.

**Reply by kirill-grouchnikov** [reply]
https://github.com/kirill-grouchnikov/radiance/commit/67645c0f772bcad558308f49fcffc43d97448543

**Reply by MichaelWDenney** [reply]
Thanks.  I'll give it a try when I have the chance to update radiance and let you know how it goes.

**Reply by MichaelWDenney** [reply]
The new `JRibbon.addAnchoredCommand(CommandButtonProjection projection, int index)` is working well to fill my need.  Thanks!
